### PR TITLE
feat: Add security config for halo

### PR DESCRIPTION
**Pull request by guqing**
### What this PR does?
为 2.0 版本增加 spring securiy 安全相关配置

Reference：
- [spring-security-mvc-integrations](https://docs.spring.io/spring-security/reference/6.0.0-M3/servlet/integrations/mvc.html)
- [spring-security-configuration](https://docs.spring.io/spring-security/reference/6.0.0-M3/servlet/configuration/java.html)
- [spring-boot-applications.with-mock-environment](https://docs.spring.io/spring-boot/docs/3.0.0-M2/reference/htmlsingle/#features.testing.spring-boot-applications.with-mock-environment)
- [spring-mvc-test-server-setup-options](https://docs.spring.io/spring-framework/docs/6.0.0-M3/reference/html/testing.html#spring-mvc-test-server-setup-options)
- [spring-security-test](https://docs.spring.io/spring-security/reference/6.0.0-M3/servlet/test/mockmvc/setup.html)

/cc @halo-dev/sig-halo
/area core

### CODE REVIEW

**Review by JohnNiang — COMMENTED** [review]
1. 尝试添加一些必要的测试
2. 尝试在 PR 描述中添加 Reviewer 需要了解的文档链接

/kind feature
/hold

File: src/main/java/run/halo/app/config/WebMvcConfig.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * @author guqing
+ * @date 2022-04-12
+ */
+@Configuration
+public class WebMvcConfig implements WebMvcConfigurer {
+
+    @Override
+    public void configurePathMatch(PathMatchConfigurer configurer) {
+        configurer.addPathPrefix("/api/v1",
+            c -> c.isAnnotationPresent(RestController.class));
```
Comment (JohnNiang) [inline src/main/java/run/halo/app/config/WebMvcConfig.java:18]: 建议使用 HandlerTypePredicate 工具，请参考：

```java
HandlerTypePredicate.forAnnotation(RestController.class)
  				.and(HandlerTypePredicate.forBasePackage("org.example"))
```

File: src/test/java/run/halo/app/integration/security/AuthenticationTest.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    @Test
+    public void allowAccessTest() throws Exception {
+        mockMvc.perform(get("/anonymous"))
+            .andDo(print())
+            .andExpect(status().is2xxSuccessful())
+            .andExpect(content().string("Hello, now you see me."));
+    }
+
+    @Test
+    public void scuredApiTest() throws Exception {
```
Comment (JohnNiang) [inline src/test/java/run/halo/app/integration/security/AuthenticationTest.java:58]:
```suggestion
    public void securedApiTest() throws Exception {
```

**Review by JohnNiang — CHANGES_REQUESTED** [review]

**Review by JohnNiang — APPROVED** [review]
/lgtm
/approve

**Review by ruibaby — APPROVED** [review]
/lgtm
/approve

**Pull request merged**